packages/data/src/components/with-select/test/index.js

```diff
@@ -0,0 +1,379 @@
+/**
+ * External dependencies
+ */
+import TestRenderer from 'react-test-renderer';
+
+/**
+ * WordPress dependencies
+ */
+import { createElement, compose } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import withSelect from '../';
+import withDispatch from '../../with-dispatch';
+import { createRegistry } from '../../../registry';
+import RegistryProvider from '../../registry-provider';
+
+describe( 'withSelect', () => {
+	let registry;
+	beforeEach( () => {
+		registry = createRegistry();
+	} );
+
+	it( 'passes the relevant data to the component', () => {
+		registry.registerStore( 'reactReducer', {
+			reducer: () => ( { reactKey: 'reactState' } ),
+			selectors: {
+				reactSelector: ( state, key ) => state[ key ],
+			},
+		} );
+
+		// In normal circumstances, the fact that we have to add an arbitrary
+		// prefix to the variable name would be concerning, and perhaps an
+		// argument that we ought to expect developer to use select from the
+		// wp.data export. But in-fact, this serves as a good deterrent for
+		// including both `withSelect` and `select` in the same scope, which
+		// shouldn't occur for a typical component, and if it did might wrongly
+		// encourage the developer to use `select` within the component itself.
```

Comment:
Bit confused by this. Are we saying that all users of `withSelect` should name the first callback argument `_select`? Why say this in a unit test and not in the README?

We already have a lint rule in Gutenberg which would prevent the `select` in `import { select } from '@wordpress/data'` from  accidentally being shadowed by `withSelect`.

Reply:
I think we're saying that we should prefer `withSelect` over `select` in most cases.

Reply:
Why name the callback argument `_select`, then?

Reply:
That makes sense! It hadn't occurred to me that we're testing both `select` and `withSelect` in these tests 🤦‍♂️